pkg/agent/agentclient/stream.go

```diff
@@ -332,6 +337,20 @@ type connectResult struct {
 	agentServiceClient agent.AgentService_ConnectClient
 }
 
+func (c *RedialableAgentClient) initializeAuthContext(ctx context.Context) (context.Context, error) {
+	var err error
+	var b []byte
+
+	// load service account's token
+	if b, err = ioutil.ReadFile(c.serviceAccountTokenPath); err != nil {
```

Comment:
I think this token won't change in the life time of the agent, then can we read the token when constructing the clientset, and pass down to the rediableAgentClient, instead of read the file every time retry connection?
I'm ok if you want to do this in a followup PR.

Reply:
It changed every 1 hour (https://kubernetes.io/docs/concepts/storage/volumes/#projected)

Reply:
Good to know. Can you add a comment to document this fact?

Reply:
done